tests/allotrope/schema_parser/model_class_editor_test.py

```diff
@@ -29,32 +33,78 @@ def validate_lines_against_multistring(class_lines: ClassLines, lines: str) -> N
     assert class_lines == class_lines_from_multistring(lines)
 
 
-def test_parse_field_types() -> None:
+def test__to_or_union() -> None:
+    assert _to_or_union({"str"}) == "str"
+    assert _to_or_union({"str", "int"}) == "int|str"
+    assert _to_or_union({"str", "None", "int"}) == "int|str|None"
+
+
+def test__is_or_union() -> None:
+    assert not _is_or_union("int")
+    assert _is_or_union("int|float")
+    assert not _is_or_union("list[int,float]")
+    assert _is_or_union("str|dict[int,float]")
+    assert _is_or_union("list[str,int]|str")
+    assert not _is_or_union("dict[float|int,str]")
+
+
+def test__split_types() -> None:
+    assert _split_types("int", ",")[0] == {"int"}
+    assert _split_types("int,str", ",")[0] == {"int", "str"}
+    assert _split_types("int|str", "|")[0] == {"int", "str"}
+    assert _split_types("list[int,float]|str", "|")[0] == {"list[int,float]", "str"}
+    assert _split_types("list[int|float],str", ",")[0] == {"list[int|float]", "str"}
+    assert _split_types("int,float]|str", ",") == ({"int", "float"}, 9)
```

Comment:
should this be `[int, float]|str`? also what's the `9` for?

Reply:
Added an explainer comment to the function and test